docs/source/reference/config.rst

```diff
@@ -103,6 +103,22 @@ Available fields and semantics:
   # Advanced GCP configurations (optional).
   # Apply to all new instances but not existing ones.
   gcp:
+    # VPC to use (optional).
+    #
+    # Default: None, which implies the following behavior. First, the VPC named
+    # 'default' is checked against minimal recommended firewall rules for
+    # SkyPilot to function. If it satisfies these rules, this VPC is used.
+    # Otherwise, a new VPC named 'skypilot-vpc' is automatically created with
+    # the minimal recommended firewall rules and will be used.
+    #
+    # If this field is set, SkyPilot will use the VPC with this name. Useful for
+    # when users want to manually set up a VPC and precisely control its
+    # firewall rules. If no region restrictions are given, SkyPilot only
+    # provisions in regions for which a subnet of this VPC exists. Errors are
+    # thrown if VPC with this name is not found. The VPC does not get modified
+    # in any way.
```

Comment:
We probably need to note down in the doc that what firewall rules are required for the VPC. Here are several rules I can think of (referring to the firewall templates in our `sky/skylet/providers/gcp/constants.py`:
1. rule for TCP on port 22, for some specific external IPs to allow ssh
2. rule for TCP on port 22, for all internal IPs to allow ssh between multiple nodes created by the user.

Reply:
It might be better to have the detail in the minimal permission doc page, as the admin may need to refer to it to create a VPC for all the users.

For the case of multi-node cluster, the ray autoscaler need the ssh connection between those nodes to setup the worker nodes from the head node.
Actually, we need not only the ssh connection among the nodes but many other ports, as the ray cluster need to connect with each other through internal ip and ports.

Reply:
Done, how does it look?

Reply:
Looks great! Thanks!